core/backend/src/TextAnnotationElement.ts

```diff
@@ -0,0 +1,106 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Elements
+ */
+
+import { GeometryParams, GeometryStreamBuilder, TextAnnotation, TextAnnotation2dProps, TextAnnotation3dProps } from "@itwin/core-common";
+import { IModelDb } from "./IModelDb";
+import { AnnotationElement2d, GraphicalElement3d } from "./Element";
+import { produceTextAnnotationGeometry } from "./TextAnnotationGeometry";
+import { Id64String } from "@itwin/core-bentley";
+
+function updateAnnotation(element: TextAnnotation2d | TextAnnotation3d, annotation: TextAnnotation, subCategory: Id64String | undefined): boolean {
+  const builder = new GeometryStreamBuilder();
+
+  const params = new GeometryParams(element.category, subCategory);
+  if (!builder.appendGeometryParamsChange(params)) {
+    return false;
+  }
+
+  const props = produceTextAnnotationGeometry({ iModel: element.iModel, annotation });
+  if (!builder.appendTextBlock(props)) {
+    return false;
+  }
+
+  element.geom = builder.geometryStream;
+  element.jsonProperties.annotation = annotation.toJSON();
+
+  return true;
+}
+
+/** An element that displays textual content within a 2d model.
+ * The text is stored as a [TextAnnotation]($common) from which the element's [geometry]($docs/learning/common/GeometryStream.md) and [Placement]($geometry) are computed.
+ * @see [[setAnnotation]] to change the textual content.
+ * @public
+ */
+export class TextAnnotation2d extends AnnotationElement2d {
+  /** @internal */
+  public static override get className(): string { return "TextAnnotation2d"; }
+  protected constructor(props: TextAnnotation2dProps, iModel: IModelDb) { super(props, iModel); }
+
+  public static fromJSON(props: TextAnnotation2dProps, iModel: IModelDb): TextAnnotation2d {
+    return new TextAnnotation2d(props, iModel);
+  }
+
+  public override toJSON(): TextAnnotation2dProps {
```

Comment:
do we need this override if all we're doing is just calling super?

Reply:
For the return type.